File: TransCoder_tokenized_test_set_functions/cpp/PRINT_N_X_N_SPIRAL_MATRIX_USING_O1_EXTRA_SPACE.cpp
void printSpiral ( int n ) {
  for ( int i = 0;
  i < n;
  i ++ ) {
    for ( int j = 0;
    j < n;
    j ++ ) {
      int x;
      x = min ( min ( i, j ), min ( n - 1 - i, n - 1 - j ) );
      if ( i <= j ) printf ( "%d\t ", ( n - 2 * x ) * ( n - 2 * x ) - ( i - x ) - ( j - x ) );
      else printf ( "%d\t ", ( n - 2 * x - 2 ) * ( n - 2 * x - 2 ) + ( i - x ) + ( j - x ) );
    }
    printf ( "\n" );
  }
}

File: TransCoder_tokenized_test_set_functions/python/PRINT_N_X_N_SPIRAL_MATRIX_USING_O1_EXTRA_SPACE.py
def printSpiral ( n ) :
    for i in range ( 0, n ) :
        for j in range ( 0, n ) :
            x = min ( min ( i, j ), min ( n - 1 - i, n - 1 - j ) )
            if ( i <= j ) :
                print ( ( n - 2 * x ) * ( n - 2 * x ) - ( i - x ) - ( j - x ), end = "\t" )
            else :
                print ( ( ( n - 2 * x - 2 ) * ( n - 2 * x - 2 ) + ( i - x ) + ( j - x ) ), end = "\t" )
        print ( )
    
